core/internal/cltest/mocks.go

```diff
@@ -213,16 +213,19 @@ func (mock *EthMock) Call(result interface{}, method string, args ...interface{}
 // Call, falling back to reflection if the values dont support the required
 // encoding interfaces
 func assignResult(result, response interface{}) error {
-	unmarshaler, uok := result.(encoding.TextUnmarshaler)
-	marshaler, mok := response.(encoding.TextMarshaler)
-
-	if uok && mok {
-		bytes, err := marshaler.MarshalText()
-		if err != nil {
-			return err
+	if unmarshaler, ok := result.(encoding.TextUnmarshaler); ok {
```

Comment:
This change allows us to unpack a `string` or `[]byte` `response`, neither of which implement `encoding.TextMarshaler`, into the `result`.

Reply:
Is a special format assumed of String or []byte in this instance?

Reply:
Was this causing issues somewhere?

Reply:
Yeah, the code in `eth.ConnectedContract#Call` that decodes results into a `hexutil.Bytes` was the  issue -- `hexutil.Bytes` is a type alias for `[]byte` with some special decoding logic.  The expected format is just regular Ethereum ABI-encoded data.